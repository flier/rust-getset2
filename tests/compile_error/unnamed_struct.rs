use getset2::Getter;

#[derive(Getter)]
pub struct UnnamedStruct(u8);

fn main() {}
