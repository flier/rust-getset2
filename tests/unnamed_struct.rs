use getset2::{Getter, Setter};

#[derive(Default, Getter, Setter)]
#[get(pub, copy)]
pub struct Unnamed(
    #[get(rename(x))]
    #[set(rename(x))]
    usize,
    usize,
);

#[test]
fn unnamed_struct() {
    let mut unnamed = Unnamed::default();

    assert_eq!(unnamed.set_x(123).x(), 123);
    assert_eq!(unnamed.set_arg1(456).arg1(), 456);
}
