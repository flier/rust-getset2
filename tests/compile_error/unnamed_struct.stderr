error: #[derive(Getter)] can only be applied to structure with named fields
 --> tests/compile_error/unnamed_struct.rs:4:1
  |
4 | pub struct UnnamedStruct(u8);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
